nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/autodiff/samediff/SameDiff.java

```diff
@@ -1496,6 +1497,33 @@ public INDArray getArr() {
 
     }
 
+    // auto naming
+
+    private Integer _var_id = 0;
+    private String getNewVarName(){
+        String varName = "sd_var_" + _var_id.toString();
+        while(variableMap.containsKey(varName)){
+            _var_id++;
+            varName = "sd_var_" + _var_id.toString();
```

Comment:
What about ```var_``` prefix instead of ```sd_var_```? It's shorter, and no less descriptive.

Reply:
Because `var_` maybe too common a prefix? For e.g:

```
x = sd.var(arr) // named var_0, user doens't know about this

// somewhere else:

x = sd.var("var_0", arrx) // name already taken
```

Reply:
Yeah, good point. OK, sounds good :)